test/tag_matcher_test.dart

```diff
@@ -0,0 +1,67 @@
+library tag_matcher_test;
+
+import '../tool/src/tag_matcher.dart';
+import 'package:test/test.dart';
+
+void main() => defineTests();
+
+void defineTests() {
+  group('TagMatcher', () {
+    TagMatcher matcher;
+    var testString = 'Some untagged <span>This is some text in a span </span>'
+        '<a href="path/to/foo">foo link</a> <span>and back to span </span>';
+
+    test('matches tag contents correctly', () {
+      matcher = TagMatcher.spanMatcher;
+      var allContents = matcher.allContents(testString);
+
+      expect(allContents.length, 2);
+      expect(allContents.first, 'This is some text in a span ');
+      expect(allContents.last, 'and back to span ');
+    });
+
+    test('matches tag contents with attributes', () {
+      matcher = TagMatcher.aMatcher;
+      var allContents = matcher.allContents(testString);
+
+      expect(allContents.length, 1);
+      expect(allContents.first, 'foo link');
+    });
+
+    test('matches attributes correctly', () {
+      matcher = TagMatcher.anyTag;
+      var allAttributes = matcher.allAttributes(testString);
+
+      expect(allAttributes.length, 3);
```

Comment:
the name of the test doesnt quite make it clear why you would expect the first and last to be empty.